### PR TITLE
Update deps.

### PR DESCRIPTION
**Pull request by dvc94ch**
This change is [Reviewable](https://reviewable.io/reviews/mullvad/system-configuration-rs/30)

Closes #22 
Closes #27 
Closes #29

### CODE REVIEW

**Comment by dvc94ch** [comment]
@faern bumped the crates and added a changelog entry

**Comment by dvc94ch** [comment]
So this is blocking a PR/release of another crate (if-watch). Would it be possible to get a new release with this PR merged?  I know it's new year and everything, so will probably take a while :)

File: system-configuration/src/dynamic_store.rs [hunk 1]
```diff
@@ -82,6 +82,7 @@ impl<T> SCDynamicStoreBuilder<T> {
     ///
     /// [`SCDynamicStore`]: struct.SCDynamicStore.html
     /// [`SCDynamicStoreCreateWithOptions`]: https://developer.apple.com/documentation/systemconfiguration/1437818-scdynamicstorecreatewithoptions?language=objc
+    #[must_use]
```
Comment (faern) [inline system-configuration/src/dynamic_store.rs:85]: Why make this `must_use`? None of the other builder methods here are `must_use`

File: system-configuration/src/network_reachability.rs [hunk 2]
```diff
@@ -178,7 +178,7 @@ impl SCNetworkReachability {
         }
 
         ReachabilityFlags::from_bits(raw_flags)
-            .ok_or_else(|| ReachabilityError::UnrecognizedFlags(raw_flags))
+            .ok_or(ReachabilityError::UnrecognizedFlags(raw_flags))
```
Comment (faern) [inline system-configuration/src/network_reachability.rs:181]: Why this change?

File: .travis.yml [hunk 3]
```diff
@@ -3,8 +3,8 @@ cache: cargo
 rust:
   - stable
   - beta
-  - nightly
-  - 1.36.0
+#  - nightly
```
Comment (faern) [inline .travis.yml:6]: Why is testing on nightly no longer being done? This looks WIP

**Review by faern — CHANGES_REQUESTED** [review]

File: system-configuration/src/dynamic_store.rs (hunk 1 above)
Comment (dvc94ch) [reply system-configuration/src/dynamic_store.rs:85]: clippy complains when returning self without must use

File: system-configuration/src/network_reachability.rs (hunk 2 above)
Comment (dvc94ch) [reply system-configuration/src/network_reachability.rs:181]: @pinkisemils did it, presumably due to clippy being obnoxious

File: .travis.yml (hunk 3 above)
Comment (dvc94ch) [reply .travis.yml:6]: Because nightly panics (which is a compiler bug). This is to be expected as it's nightly.

Comment (faern) [reply .travis.yml:6]: I'm not expecting that. I have been building my crates on nightly in CI for years and very rarely have issues. But OK. I can't find the travis log right now.

File: system-configuration/src/dynamic_store.rs (hunk 1 above)
Comment (faern) [reply system-configuration/src/dynamic_store.rs:85]: Ah. This is a nightly lint. Our CI setup on this crate is so terribly outdated. We should not run clippy on all rust versions. Would be great to have this crate migrated to github actions.

**Comment by faern** [comment]
I'm not a fan of version bumps in PRs not specifically dedicated to making releases. And this PR changes the version twice. The way we'll proceed is

1. Upgrade the CI to github actions and make it not error on nighlty specific clippy lints: #31 
2. Rebase and remove the version bumps from the existing core-foundation bump PR: #29 
3. Bump the version and make a release separately.

**Comment by dvc94ch** [comment]
Sounds good. The reason for the version bump was you requested it previously in the review, I usually let maintainers do it unless it's requested. Ping me if you want me to do something, looks like you're taking care of it now.

**Comment by faern** [comment]
This PR has been made obsolete by its predecessor #29. But thanks for the push that made us do this. I hope I can get a new release out fairly soon. Hope it works well for you!

**Comment by faern** [comment]
Version 0.5.0 has been released with the new `core-foundation`
